fundingmanager.go

```diff
@@ -585,21 +708,226 @@ func (f *fundingManager) handleFundingComplete(fmsg *fundingCompleteMsg) {
+	// the state to disk as we can now open the channel.
+	commitSig := fmsg.msg.CommitSignature.Serialize()
+	completeChan, err := resCtx.reservation.CompleteReservation(nil, commitSig)
+	if err != nil {
+		fndgLog.Errorf("unable to complete reservation sign complete: %v", err)
+		resCtx.err <- err
+		return
+	}
+
+	fundingPoint := resCtx.reservation.FundingOutpoint()
+	fndgLog.Infof("Finalizing pendingID(%v) over ChannelPoint(%v), "+
+		"waiting for channel open on-chain", chanID, fundingPoint)
+
+	// Send an update to the upstream client that the negotiation process
+	// is over.
+	// TODO(roasbeef): add abstraction over updates to accommodate
+	// long-polling, or SSE, etc.
+	resCtx.updates <- &lnrpc.OpenStatusUpdate{
+		Update: &lnrpc.OpenStatusUpdate_ChanPending{
+			ChanPending: &lnrpc.PendingUpdate{
+				Txid:        fundingPoint.Hash[:],
+				OutputIndex: fundingPoint.Index,
+			},
+		},
+	}
+
+	go func() {
+		doneChan := make(chan struct{})
+		go f.waitForFundingConfirmation(completeChan, doneChan)
+
+		<-doneChan
+
+		// Finally give the caller a final update notifying them that
+		// the channel is now open.
+		// TODO(roasbeef): helper funcs for proto construction
+		resCtx.updates <- &lnrpc.OpenStatusUpdate{
+			Update: &lnrpc.OpenStatusUpdate_ChanOpen{
+				ChanOpen: &lnrpc.ChannelOpenUpdate{
+					ChannelPoint: &lnrpc.ChannelPoint{
+						FundingTxid: fundingPoint.Hash[:],
+						OutputIndex: fundingPoint.Index,
+					},
+				},
+			},
+		}
+
+		f.deleteReservationCtx(peerKey, fmsg.msg.ChannelID)
+	}()
+}
+
+// waitForFundingConfirmation handles the final stages of the channel funding
+// process once the funding transaction has been broadcast. The primary
+// function of waitForFundingConfirmation is to wait for blockchain
+// confirmation, and then to notify the other systems that must be notified
+// when a channel has become active for lightning transactions.
+func (f *fundingManager) waitForFundingConfirmation(
+	completeChan *channeldb.OpenChannel, doneChan chan struct{}) {
+
+	// Register with the ChainNotifier for a notification once the funding
+	// transaction reaches `numConfs` confirmations.
+	txid := completeChan.FundingOutpoint.Hash
+	numConfs := uint32(completeChan.NumConfsRequired)
+	confNtfn, _ := f.cfg.Notifier.RegisterConfirmationsNtfn(&txid, numConfs)
+
+	fndgLog.Infof("Waiting for funding tx (%v) to reach %v confirmations",
+		txid, numConfs)
+
+	// Wait until the specified number of confirmations has been reached,
+	// or the wallet signals a shutdown.
+	confDetails := <-confNtfn.Confirmed
+
+	fundingPoint := completeChan.FundingOutpoint
+	fndgLog.Infof("ChannelPoint(%v) is now active",
+		fundingPoint)
+
+	completeChan.IsPending = false
+	err := f.cfg.Wallet.ChannelDB.MarkChannelAsOpen(fundingPoint)
+	if err != nil {
+		fndgLog.Errorf("error setting channel pending flag to false: "+
+			"%v", err)
+		return
+	}
+
+	// Finally, create and officially open the payment channel!
```

Comment:
We wouldn't need to pass in the `Signer` nor `Notifier` if `CompleteReservation` returned a whole `lnwallet.LightningChannel` directly.